cpplint.py

```diff
@@ -52,10 +52,20 @@
 import sys
 import unicodedata
 
+# Files with any of these extensions are considered to be
+# header files (and will undergo different style checks).
+# This set can be extended by using the --headers
+# option (also supported in CPPLINT.cfg)
+_header_extensions = ['h', 'hpp', 'hxx', 'h++', 'cuh']
+_nonheader_extensions = ['c', 'cc', 'cpp', 'cxx', 'c++', 'cu']
+
+
 # The allowed extensions for file names
 # This is set by --extensions flag.
-_valid_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'h', 'hpp', 'hxx',
-    'h++'])
+_valid_extensions = set(_nonheader_extensions + _header_extensions)
+
+# files with this suffix before the extension will be treated as test files
+_test_suffixes = set(['_unittest', '_test', '_regtest'])
```

Comment:
so, that may be a nice additional feature, but maybe that would go well into #11